src/js/filter.js

```diff
@@ -0,0 +1,39 @@
+/* global document, window */
+
+const setUpFilter = (markerGroup, citiesToMarkers, data) => {
+  const filter = document.querySelector(".filter--scope");
+  let priorSelection = [
+    "Regional",
+    "Citywide",
+    "City Center",
+    "Transit Oriented",
+    "Main Street",
+  ];
+
+  const onScopeFilterChange = (markerGroup, citiesToMarkers, data) => () => {
+    const checked = document.querySelectorAll(".filter--scope :checked");
+    const selected = [...checked].map((option) => option.value);
+    Object.entries(citiesToMarkers).forEach(([cityState, marker]) => {
+      if (
+        selected.some((scope) =>
+          data[cityState]["report_magnitude"].includes(scope)
+        )
```

Comment:
We can make this more efficient. We want to know if there is an intersection between two sets: are any elements in both collections? 

Right now, both data types are an Array<string>. To check if an element is an array is `O(n)`, that you have to in the worst case iterate through the entire array. That means you have time complexity `O(s * c)`, where `s` is the number of selected scopes and `c` is the number of report_magnitudes belonging to the city.

In contrast, Sets and Dictionaries/HashMaps are `O(1)` to see if an element is contained, rather than `O(n)`.

So, you can rewrite change your `selected`/`checked ` variable to be a `Set` data structure. Then, check if any element of `data[cityState]["report_magnitude"]` is included in that set:

```javascript
if (
  data[cityState]["report_magnitude"].some(
    (reportMagnitude) => checked.has(reportMagnitude)
   )
)
```

This is `O(c)`, since you've replaced the `O(s)` portion with `O(1)`. `O(c * 1)` becomes `O(c)`.

--

It's not a big deal because `c` and `s` are both such small numbers here. But still a good win for little extra complexity.